utils/tinystr/src/ascii.rs

```diff
@@ -110,30 +114,33 @@ impl<const N: usize> TinyAsciiStr<N> {
     #[must_use]
     pub fn len(&self) -> usize {
         if N <= 4 {
-            Aligned4::from_bytes(&self.bytes).len()
+            Aligned4::from_asciibytes(&self.bytes).len()
         } else if N <= 8 {
-            Aligned8::from_bytes(&self.bytes).len()
+            Aligned8::from_asciibytes(&self.bytes).len()
         } else {
-            self.bytes.iter().position(|x| *x == 0).unwrap_or(N)
+            self.bytes
+                .iter()
+                .position(|x| *x == AsciiByte::B0)
+                .unwrap_or(N)
         }
     }
 
     #[inline]
     #[must_use]
     pub const fn is_empty(&self) -> bool {
-        self.bytes[0] == 0
+        self.bytes[0] as u8 == AsciiByte::B0 as u8
     }
 
     #[inline]
     #[must_use]
     pub fn as_bytes(&self) -> &[u8] {
-        &self.bytes[0..self.len()]
+        unsafe { core::mem::transmute(&self.bytes[0..self.len()]) }
```

Comment:
Annotate each unsafe block with why it's safe

Reply:
Done.